internal/vcs/metadata.go

```diff
@@ -270,11 +655,57 @@ type Pipeline struct {
 	ID string
 }
 
-// Metadata holds a snapshot of information for a given environment and vcs system.
+// Remote holds information about the upstream repository that the git project uses.
+type Remote struct {
+	Host string
+	URL  string
+}
+
+// Metadata holds a snapshot of information for a given environment and VCS system.
 // PullRequest and Pipeline properties are only populated if running in a CI system.
 type Metadata struct {
+	Remote      Remote
 	Branch      Branch
 	Commit      Commit
 	PullRequest *PullRequest
 	Pipeline    *Pipeline
 }
+
+func isMergeCommit(message string) bool {
+	return strings.Contains(strings.ToLower(message), "merge")
```

Comment:
This can provide false positives for commit messages like "feat: Merge X and Y into Z". Should we use a regex that ensures that the message starts with `/^merge/i`?

Another alternative might be checking commit's parents - merge commit should have multiple. So if we have `c` as go-git's commit struct, `len(c.ParentHashes()) > 1` would mean it's a merge commit (in theory, I haven't checked myself). What do you think?

Reply:
Good shout with `regexp`. Yeah I tried using the `c.ParentHashes` approach but this seemed to be inconsistent as to which was the hash of the `source` and the `target` branch, so I skipped it.

Reply:
resolved in https://github.com/infracost/infracost/pull/1920/commits/15c83f1965286b7e7cc6e9e06dd19cec9a8fa136